### ISSUE TITLE
Non intuitive behaviour when adding to a selection via a Command/Shift/Ctrl

### ISSUE THREAD

**Issue by chipatooth** [ThibaultJanBeyer/DragSelect#9]
Nice library but I found one thing rather odd. I'm not sure if this is a bug or by design but I find that when I select an item then, hold down Ctrl or Command and drag select it deselects the original item if I drag over it.

You can reproduce this on your examples page:

1. Select any box (say #4)
2. Hold down Ctrl/Command
3. Drag select over all boxes including #4.

What happens:
1-3, 5-6 are selected. 4 is deselected.

What I would expect to happen:
1-6 are selected.

If this is a bug, please let me know if could fix it.
If this is by design: is it possible to make this an option. I would like the option to basically keep selected whatever was selected before the second (or nth) on drag was started.

Maybe different configurable keys for drag to add, drag to remove, drag for inverse selection. Not sure because I'm not sure of the original intent. I have some people scratching their heads over the current behaviour.

Thanks ahead for any time you can spare.

**Comment by ThibaultJanBeyer** [comment]
It is exactly the same behaviour as on native operating systems when using ctrl/cmd to select files. (at least on OSX)

However, the idea, with adding an option to turn that functionality off is probably a good idea.
If you feel confident you could give it a try and make a pull request. If not, I‘ll work on it asap. 
Just please let me know.

**Comment by chipatooth** [comment]
_"It is exactly the same behaviour as on native operating systems when using ctrl/cmd to select files. (at least on OSX)"_

That's kind of true: finder remembers your previous selection during a command+drag. What I mean is:

1. Open finder on OSX in a folder that has a bunch of files
2. Single select a file somewhere in the middle
3. Now somewhere below that file start a Command Drag up over that file (don't let go)
4. So, like you describe, it should select the files below the first file and inverse the selection of that file.
5. Now drag your selection down below the file again. (assuming you didn't let go)
6. The file is now restored to being selected and the other files are deselected.

Basically Finder snapshots what was selected before command+drag and reverts back as you move the drag area without committing. Your library loses the initial selection.

This, in addition to a union vs intersection select option, would probably need to be fixed/optioned-in. 

I'm not familiar enough with your codebase to know how difficult it would be to make the changes. I'll take a look next week if I have time. If you can think of anything in the meantime: I appreciate it.

**Comment by ThibaultJanBeyer** [comment]
Hmm, that sounds more like a bug… Pardon me, but I’m still not sure that I understood it correctly, could you possibly make a video of the behavior with DragSelect tool?
So that I can reproduce it? That would be extremely kind.

For your convenience, you can use one of the Codepen examples, for example, this one: https://codepen.io/ThibaultJanBeyer/pen/prpwYG

On a Mac you can use "Quick Time Player" then file > new screen recording.
Otherwise some other free screen capture method.

**Comment by ThibaultJanBeyer** [comment]
Ah! I think I understood, when you selected elements, then select new elements with CTRL, hover an element then move out of that element the element is still selected, however it should be unselected? It’s quite hard to explain yes. Am I right?

If so, then yes, it seems more like a usability bug. If what I described is what you mean, then no need for a video. I’ll have a look into it as soon as possible. Unless you’re already working on a solution.

Thanks for reporting!

**Comment by ThibaultJanBeyer** [comment]
Happy to announce that this is solved in version 1.7.17. Related commit is https://github.com/ThibaultJanBeyer/DragSelect/commit/c9f85752ca82c7a5ad5ddaa18c47c3fdcef4846a
@chipatooth please update to the newest version and let me know if everything is as expected now.
Thanks again for using the tool and reporting issues 💯
